pkg/util/identifiable/identifiable.go

```diff
@@ -0,0 +1,74 @@
+package identifiable
+
+import (
+	"reflect"
+)
+
+// Identifiable is a simple interface wrapping any object which has some key field which can be used for later
+// aggregation operations (grouping, intersection, difference etc)
+type Identifiable interface {
+	Identifier() interface{}
+}
+
+// SetDifference returns all 'Identifiable' elements that are in left slice and not in the right one
+func SetDifference(left, right []Identifiable) []Identifiable {
+	result := make([]Identifiable, 0)
+	for _, l := range left {
+		found := false
+		for _, r := range right {
+			if r.Identifier() == l.Identifier() {
+				found = true
+				break
+			}
+		}
+		if !found {
+			result = append(result, l)
+		}
+	}
+	return result
+}
+
+// SetIntersection returns all 'Identifiable' elements from 'left' and 'right' slice that intersect by 'Identifier()'
+//value. Each intersection is represented as a tuple of two elements - matching elements from 'left' and 'right'
+func SetIntersection(left, right []Identifiable) [][]Identifiable {
+	result := make([][]Identifiable, 0)
+	for _, l := range left {
+		for _, r := range right {
+			if r.Identifier() == l.Identifier() {
+				result = append(result, []Identifiable{l, r})
+			}
+		}
+	}
+	return result
+}
+
+// SetDifferenceGeneric is a convenience function solving lack of covariance in Go: it allows to pass the arrays declared
+// as some types implementing 'Identifiable' and find the difference between them
+// Important: the arrays past must declare types implementing 'Identifiable'!
+func SetDifferenceGeneric(left, right interface{}) []Identifiable {
```

Comment:
yes, I think I agree with you.
All the code will try to declare the array with a specific type (e.g. `[]DBUser`) instead of the interface one (`[]Indentifiable`) so the first method won't get used at all.
We've seen this in Enterprise Operator code.
Will rename the methods 👍  and stop exporting non-generic ones